src/azure-cli/azure/cli/command_modules/acs/agentpool_decorator.py

```diff
@@ -931,6 +931,26 @@ def get_max_surge(self):
         # this parameter does not need validation
         return max_surge
 
+    def get_drain_timeout(self):
+        """Obtain the value of drain_timeout.
+
+        :return: int
+        """
+        # read the original value passed by the command
+        drain_timeout = self.raw_param.get("drain_timeout")
+        # In create mode, try to read the property value corresponding to the parameter from the `agentpool` object
+        if self.decorator_mode == DecoratorMode.CREATE:
+            if (
+                self.agentpool and
+                self.agentpool.upgrade_settings and
+                self.agentpool.upgrade_settings.drain_timeout is not None
```

Comment:
@paulgmiller @FumingZhang ^^^ shouldn't that be

```
self.agentpool.upgrade_settings.drain_timeout_in_minutes is not None
```

?

Reply:
Sigh I miss static typing.